packages/js-commons/src/core/local-files.ts

```diff
@@ -253,6 +259,12 @@ export const localStoreAsset = async (
   const { relativePath } = await saveArtifact(assetDocument, appMetadata, {
     relativePath: fileName,
   });
+
+  if (options?.skipGlobalMetadata !== true) {
+    merge(appMetadata.manifest, { [assetDocument.id]: assetDocument });
+    appsMetadata[appId] = appMetadata;
+    await setGlobalMetadata(appsMetadata);
+  }
```

Comment:
Why do you need to make changes to the global metadata here?

Reply:
to reflect latest manifest of the app